File: src/main/java/com/trade/securities/adapter/in/web/EtfTradingInfoController.java
package com.trade.securities.adapter.in.web;

import com.trade.securities.application.port.in.GetEtfTradingInfoUseCase;
import com.trade.securities.domain.EtfTradingInfo;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/etf")
@RequiredArgsConstructor
public class EtfTradingInfoController {

    private final GetEtfTradingInfoUseCase getEtfTradingInfoUseCase;

    @GetMapping("/trading-info")
    public List<EtfTradingInfo> getEtfTradingInfo(@RequestParam String basDd) {
        return getEtfTradingInfoUseCase.getEtfTradingInfo(basDd);
    }
}

File: src/main/kotlin/com/trade/securities/adapter/in/web/EtfTradingInfoController.kt
package com.trade.securities.adapter.`in`.web

import com.trade.securities.application.port.`in`.GetEtfTradingInfoUseCase
import com.trade.securities.domain.EtfTradingInfo
import jakarta.enterprise.context.ApplicationScoped
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.QueryParam
import jakarta.ws.rs.core.MediaType

@ApplicationScoped
@Path("/api/etf")
@Produces(MediaType.APPLICATION_JSON)
class EtfTradingInfoController(
    private val getEtfTradingInfoUseCase: GetEtfTradingInfoUseCase
) {

    @GET
    @Path("/trading-info")
    fun getEtfTradingInfo(@QueryParam("basDd") basDd: String): List<EtfTradingInfo> {
        return getEtfTradingInfoUseCase.getEtfTradingInfo(basDd)
    }
}
